source/sync/migrate-sync-modes.txt

```diff
@@ -0,0 +1,356 @@
+   enable Flexible Sync in the future.
+
+.. _realm-sync-migrate-commit:
+
+Commit Migration
+~~~~~~~~~~~~~~~~
+
+After your migration has completed, it is in an evaluation state. You can manually
+commit the migration to make it permanent or revert the migration to return your
+App to Partiton-Based Sync.
+
+If you don't commit or revert the migration, it will automatically commit
+according to your 
+:atlas:`minimum oplog window </cluster-additional-settings/#set-minimum-oplog-window>`. 
+For example, if your minimum oplog window is 48 hours, you will have 48 hours to
+revert your migration.
+
+After a migration is committed, the old Partition-Based Sync metadata is deleted
+and you cannot revert the migration.
+
+**To manually commit a migration:** 
+
+#. In the App Services UI, go to your App and select the Device Sync page.
+#. At the top of the page, select the three dots next to 
+   :guilabel:`Manage Sync Migration`.
+#. From the list of options, select :guilabel:`Commit migration to Flexible Sync`.
+   
+   .. image:: /images/commit-migration.png
+      :alt: Commit a migration
+#. Confirm migration commit by typing "Commit", then select :guilabel:`Commit`.
+   
+   .. image:: /images/confirm-migration-commit.png
+      :alt: Confirm migration commit
+   
+   This will commit your migration and you will not be able to recover your
+   Partition-Based Sync Metadata.
+
+.. _realm-sync-migrate-revert:
+
+Revert Migration
+~~~~~~~~~~~~~~~~
+
+You can revert a migration after it has completed and before it has been committed.
+
+Completed migrations are automatically committed in accordance with your 
+:atlas:`minimum oplog window </atlas/cluster-additional-settings/#set-minimum-oplog-window>`.
+You can increase or decrease the amount of time you have to 
+revert a migration by adjusting the minimum oplog window.
+
+You cannot revert a migration that has been committed.
+
+Similar to canceling a migration, if you revert a migration, you need to 
+start over from the beginning with future migration attempts.
+
+**To revert a migration:** 
+
+#. In the App Services UI, go to your App and select the Device Sync page.
+#. At the top of the page, select the three dots next to 
+   :guilabel:`Manage Sync Migration`.
+#. From the list of options, select 
+   :guilabel:`Revert back to Partition-Based Sync`.
+   
+   .. image:: /images/revert-migration.png
+      :alt: Revert a migration
+   
+   This will revert your migration. You will need to migrate again if you 
+   want to enable Flexible Sync in the future.
+
+.. _realm-sync-migrate-after:
+
+After Backend Migration
+-----------------------
+
+Your migrated backend uses a partition key field in your Realm object models
+to map the Partition-Based Sync client objects to Flexible Sync equivalents on
+the backend. If your object models don't have a partition key field, the
+backend automatically injects one into every object the Partition-Based Sync
+client creates.
```

Comment:
You might want to point out that this value will no longer be injected once the client app has been migrated to flexible sync, which could lead to the PBS client apps not seeing the new data. The workaround is to include the partition key field in the object model and manually fill in this value if the data should be seen by PBS client apps.

Reply:
@michael-wb, I'm not sure I follow.
> once the client app has been updated to use flexible sync
> which could lead to the PBS client apps not seeing the new data

Both can't be true, can they? The PBS clients should update to Flexible Sync.

Reply:
Hopefully this is clearer:

When a PBS client app has migrated to use flexible sync, the partition field will be populated with the partition value automatically by the server. The PBS client app in the migrated state relies on this partition value to subscribe to the data to download from the server.
One the client app has been updated to natively use flexible sync (i.e., no longer a PBS app), the server will no longer automatically populate the partition field. Any PBS versions of the client app still out in the wild will not see/download any objects created by the FLX version of the client app unless the partition field is manually populated when the object is created.

Reply:
It is not guaranteed that users will update their apps (at the same time or ever) to the new FLX version.

Reply:
OK, I was wondering if that was the case. Thank you for clarifying!